### PR TITLE
Batched Dependabot updates (includes Storybook major upgrade)

### PR DESCRIPTION
**Pull request by elenagarrone**
This PR was created by the [Dependabot Batcher](https://github.com/Legal-and-General/dependabot-batcher) Action by combining the following PRs:

**Mon Jan 30 2023**

- #1143
- #1142
- #1141
- #1139

**Mon Feb 06 2023**

- #1149
- #1148
- #1146

**Wed Mar 15 2023**

- #1165

**Wed Apr 26 2023**

- #1186

**Wed May 24 2023**

- #1197

**Mon Jul 10 2023**

- #1210

**Wed Jul 19 2023**

- #1213

**Wed Aug 09 2023**

- Run `npm update` and `npm audit fix`
- Run the `build:icons` script to apply the changes brought in by `svg-to-ts` and `svgo` (attributes are now ordered)
- **Update Storybook to `v7.2.2` and fixes all its breaking changes.**

### CODE REVIEW

**Review by paul-request — APPROVED** [review]

File: .stylelintrc [hunk 1]
```diff
@@ -12,6 +12,7 @@
       }
     ],
     "at-rule-no-unknown": null,
+    "media-query-no-invalid": null,
```
Comment (elenagarrone) [inline .stylelintrc:15]: This fails for media queries that use scss

File: package.json [hunk 2]
```diff
@@ -39,95 +39,96 @@
...
+    "prettier": "^3.0.1",
     "pretty-quick": "^3.1.3",
     "protractor": "^7.0.0",
-    "sass": "^1.56.1",
-    "semantic-release": "^19.0.5",
-    "stylelint": "^14.15.0",
-    "stylelint-config-recommended": "^9.0.0",
-    "stylelint-selector-bem-pattern": "^2.1.1",
-    "svg-to-ts": "^8.9.0",
+    "react": "^18.2.0",
```
Comment (elenagarrone) [inline package.json:118]: `react` and `react-dom` are now required by Stroybook.

File: package.json [hunk 3]
```diff
@@ -39,95 +39,97 @@
...
+    "react-dom": "^18.2.0",
+    "remark-gfm": "^3.0.1",
```
Comment (elenagarrone) [inline package.json:120]: This package had to be added as MDX doesn't support GitHub Flavoured Markdown anymore: https://storybook.js.org/docs/react/writing-docs/mdx#lack-of-github-flavored-markdown-gfm

File: package.json [hunk 4]
```diff
@@ -39,95 +39,97 @@
...
+    "@angular/compiler-cli": "^14.3.0",
+    "@angular/language-service": "^14.3.0",
+    "@commitlint/cli": "^17.6.7",
+    "@commitlint/config-angular": "^17.6.7",
+    "@commitlint/config-conventional": "^17.6.7",
+    "@commitlint/prompt": "^17.6.7",
+    "@compodoc/compodoc": "^1.1.21",
+    "@semantic-release/github": "^9.0.4",
+    "@semantic-release/npm": "^9.0.2",
+    "@storybook/addon-a11y": "^7.2.2-alpha.0",
```
Comment (elenagarrone) [inline package.json:74]: `7.2.2-alpha.0` is the version installed by the migration tool.

File: .storybook/preview-head.html [hunk 5]
```diff
@@ -9,14 +9,14 @@
   }
 
   /* Custom styling to override Storybook docs global styles */
-  .sbdocs-h1 {
+  .sbdocs h1.css-wzniqs {
```
Comment (Xinchro) [inline .storybook/preview-head.html:12]: what is `css-wzniqs`?

**Review by pauleustice — APPROVED** [review]
Bit of a slog to review but well done :D

**Review by Xinchro — APPROVED** [review]

File: .storybook/preview-head.html (hunk 5 above)
Comment (elenagarrone) [reply .storybook/preview-head.html:12]: It's a class that Storybook adds to elements within the docs to apply their default styles.

**Pull request merged**